lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -316,6 +335,11 @@ static std::string getTypeName(Location loc, Type type) {
       typeName += "_si" + std::to_string(type.getIntOrFloatBitWidth());
     else
       typeName += "_ui" + std::to_string(type.getIntOrFloatBitWidth());
+  } else if (type.isa<TupleType>()) {
+    auto tupleType = type.dyn_cast<TupleType>();
+    typeName += "_tuple";
+    for (auto elementType : tupleType.getTypes())
+      typeName += getTypeName(loc, elementType);
```

Comment:
~~Should this be interleaved with `_`?~~ i see that `getTypeName` returns a `_` prefixed name... bad! But let's leave that up to a follow-up commit.